File: POSWeb/wwwroot/js/labor-costs-duration.js
$(document).ready(function () {
    const spinner = $('.choose-estimates .spinner-border');

    $('#labor-costs-duration-btn').click(function () {
        spinner.addClass('d-inline-block');
        let formData = new FormData();
        AppendEstimateFiles(formData);
        AppendLaborCostsDurationVM(formData);
        $.ajax({
            url: '/LaborCostsDuration/WriteLaborCostsDuration',
            data: formData,
            processData: false,
            contentType: false,
            type: 'POST',
            success: function () {
                spinner.removeClass('d-inline-block');
                window.location = '/LaborCostsDuration/Download';
            }
        });
    });

    function AppendLaborCostsDurationVM(formData) {
        formData.append('NumberOfWorkingDays', ReplaceDotWithComma($('#number-of-working-days').val()));
        formData.append('WorkingDayDuration', ReplaceDotWithComma($('#working-day-duration').val()));
        formData.append('Shift', ReplaceDotWithComma($('#shift').val()));
        formData.append('NumberOfEmployees', $('#number-of-employees').val());
        formData.append('AcceptanceTimeIncluded', $('#acceptance-time-included').is(":checked"));
        formData.append('TechnologicalLaborCosts', ReplaceDotWithComma($('#technological-labor-costs').val()));
    }
});

File: POSCore/DurationLogic/LaborCosts/LaborCostsDuration.cs
namespace POSCore.DurationLogic.LaborCosts
{
    public class LaborCostsDuration
    {
        public decimal Duration { get; set; }
        public decimal LaborCosts { get; }
        public decimal WorkingDayDuration { get; }
        public decimal Shift { get; }
        public decimal NumberOfWorkingDays { get; }
        public int NumberOfEmployees { get; }
        public decimal RoundedDuration { get; }
        public decimal TotalDuration { get; }
        public decimal PreparatoryPeriod { get; }
        public decimal AcceptanceTime { get; }
        public bool AcceptanceTimeIncluded { get; }
        public bool RoundingIncluded { get; }

        public LaborCostsDuration(decimal duration, decimal laborCosts, decimal workingDayDuration, decimal shift, 
            decimal numberOfWorkingDaysInMonth, int numberOfEmployees, decimal totalDuration, decimal preparatoryPeriod,
            decimal roundedDuration, decimal acceptanceTime, bool acceptanceTimeIncluded, bool roundingIncluded)
        {
            Duration = duration;
            LaborCosts = laborCosts;
            WorkingDayDuration = workingDayDuration;
            Shift = shift;
            NumberOfWorkingDays = numberOfWorkingDaysInMonth;
            NumberOfEmployees = numberOfEmployees;
            RoundedDuration = roundedDuration;
            TotalDuration = totalDuration;
            PreparatoryPeriod = preparatoryPeriod;
            AcceptanceTime = acceptanceTime;
            AcceptanceTimeIncluded = acceptanceTimeIncluded;
            RoundingIncluded = roundingIncluded;
        }
    }
}
